liblumen_alloc/src/erts/term/convert.rs

```diff
@@ -32,6 +30,14 @@ impl From<Infallible> for BoolError {
     }
 }
 
+#[derive(Debug, Error)]
+pub enum TryIntoByteVecError {
```

Comment:
This is a duplicate of the type in `maybe_aligned_maybe_binary`. This is another case where I think having different errors (type error and then binary-specific errors) and wrapping them up in something like `anyhow::Error` is a better approach; in other words, use `TypeError` and `InvalidBinaryError` independently, and anywhere that multiple errors may need to be returned, use `anyhow:Error` to unify them. With that approach we end up needing a lot less one-off types like this, and it is just ergonomically a lot more pleasant when you can return "real" error types and then add on additional context as needed.